packages/core/src/molecule/codec.advanced.ts

```diff
@@ -0,0 +1,373 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+
+import { bytesConcat, bytesFrom, BytesLike } from "../bytes/index.js";
+import { bytes, Codec, EncodableType, fixedBytes } from "./codec.js";
+import { Uint32LE } from "./predefined.js";
+
+/**
+ * Vector with fixed size item codec
+ * @param itemCodec fixed-size vector item codec
+ */
+export function fixedItemVec<Encodable>(
+  itemCodec: Codec<Encodable>,
+): Codec<Array<Encodable>> {
+  if (itemCodec.byteLength === undefined) {
+    throw new Error("fixedItemVec: itemCodec requires a byte length");
+  }
+  return bytes({
+    encode(userDefinedItems) {
+      const encodedArray = userDefinedItems.map((item) =>
+        itemCodec.encode(item),
+      );
+      return bytesConcat(
+        Uint32LE.encode(BigInt(userDefinedItems.length)),
+        encodedArray.reduce(
+          (concated, item) => bytesConcat(concated, item),
+          new ArrayBuffer(0),
```

Comment:
`Bytes` is `Uint8Array`, `new Uint8Array([])` is more precise or `[]` is simpler.